packages/react-emotion/src/index.js

```diff
@@ -90,6 +90,12 @@ export default function(tag, options: { e: string }) {
     Styled.__emotion_styles = styles
     Styled.__emotion_base = baseTag
 
+    Styled.withComponent = nextTag => {
+      return createStyled(nextTag, options)(styles)
```

Comment:
👍 this is much cleaner and more obvious from first glance. Thanks @mitchellhamilton .